src/main/java/soonwee/duke/Deadline.java

```diff
@@ -0,0 +1,30 @@
+package soonwee.duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a Deadline instance. A Deadline instance will store the task
+ * name and its due completion time.
+ */
+public class Deadline extends Task {
+
+    protected LocalDateTime by;
```

Comment:
Perhaps you could consider using a more specific name such as `dateBy` so it is clearer to readers?